### PR TITLE
feat: projects limit

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

Limit number of projects in the backend to 500.

Next PR:
* disable add project button when the limit is reached and prevent new projects in the UI

## Discussion points

Enforcing minimum limit of 1 project. This is to prevent configuration mistakes where no projects are allowed.

### CODE REVIEW

File: src/lib/openapi/spec/resource-limits-schema.ts [hunk 1]
```diff
@@ -82,6 +83,12 @@ export const resourceLimitsSchema = {
             example: 50,
             description: 'The maximum number of environments allowed.',
         },
+        projects: {
+            type: 'integer',
+            minimum: 1,
```
Comment (kwasniew) [inline src/lib/openapi/spec/resource-limits-schema.ts:88]: this is also enforced in the service

File: src/lib/features/project/project-service.ts [hunk 2]
```diff
@@ -304,6 +311,18 @@ export default class ProjectService {
             await this.validateEnvironmentsExist(environments);
         }
     }
+
+    async validateProjectLimit() {
+        if (!this.flagResolver.isEnabled('resourceLimits')) return;
+
+        const limit = Math.max(this.resourceLimits.projects, 1);
```
Comment (thomasheartman) [inline src/lib/features/project/project-service.ts:318]: Any particular reason why you wanna put this minimum here and not in the parsing of the value?

File: src/lib/openapi/spec/resource-limits-schema.ts (hunk 1 above)
Comment (thomasheartman) [reply src/lib/openapi/spec/resource-limits-schema.ts:88]: I'm sure you know this, but the limit here really has no bearing on what goes on in Unleash, right? This is just a promise we make to our end users. And in fact, in this case, if you set it to 0, you'll get 0 out. Unleash will say the schema doesn't allow the value, but it won't correct it for you.

Unless I'm missing something?

**Review by thomasheartman — APPROVED** [review]
Nice one! Just a few minor comments.

File: src/lib/features/project/project-service.ts (hunk 2 above)
Comment (kwasniew) [reply src/lib/features/project/project-service.ts:318]: Will do this in the next PR. I like your idea more.

File: src/lib/openapi/spec/resource-limits-schema.ts (hunk 1 above)
Comment (kwasniew) [reply src/lib/openapi/spec/resource-limits-schema.ts:88]: I wanted to communicate that you'll never get 0 in the GET request for UI config with the resource limits. The parsing (`Math.max(1, parseEnvVarNumber(process.env.UNLEASH_PROJECTS_LIMIT, 500))`) we're doing should provide this guarantee

**Comment by thomasheartman** [comment]
Side note: should we consider setting minimum values in parsing for all the limits? So that you can't set nagitev ilimts etc?

**Pull request merged**